### DISCUSSION TITLE
Reconsider emacs dependency for brew formula

### DISCUSSION THREAD

**Discussion by indigoviolet** [kaushalmodi/eless#43] [Q&A]
I have a separate emacs installation and don't want to use `brew`'s installation; I wonder how common that is and whether `emacs` should be a required dependency in the formula.

**Comment by kaushalmodi** [comment]
> don't want to use brew's installation

I don't have a macOS machine, so I don't use brew personally. I had added that through a user request and PR.

If you don't want to use brew, you can follow these instructions: https://eless.scripter.co/#installation

> whether emacs should be a required dependency in the formula.

Well, eless basically is a shell script that calls emacs in batch mode. So it makes sense to have emacs as a dependency.

**Comment by indigoviolet** [comment]
Thanks for the quick response. My reasoning was that a user (who has discovered and is capable of using `eless`) can easily `brew install emacs` themselves if they wanted; it is more of a pain to figure out how not to install that dependency (`--ignore-dependencies`) or blow away their existing installation. In any case, if you're not the author/owner of the formula, this issue can be closed.

**Comment by kaushalmodi** [comment]
@raxod502 If you have time, can you address @indigoviolet 's concern here? Should emacs be removed as a dependency from eless' brew recipe?

**Reply by raxod502** [reply]
No, IMO this is something that should be addressed in the package manager, not in individual packages. The package manager must provide a mechanism for you to say you have installed a package manually, and that it should be considered to satisfy dependencies for that package. For example, if you were to install some other software that depended on Emacs, you would surely want your custom installation to satisfy that requirement as well, right? The solution is not to modify all software everywhere to remove dependencies on any package that might be installed manually; it is for the end user to use Homebrew's feature of setting a package as manually installed---or, if such a feature doesn't exist, submitting a feature request upstream.

**Reply by indigoviolet** [reply]
I generally agree that Homebrew should provide this (some [relevant discussion](https://github.com/Homebrew/brew/issues/10754#issuecomment-789017245)); I have also already solved this for myself so I don't feel too strongly. But just to play devil's advocate: we haven't indicated `perl`, `bash` and `texinfo` as dependencies, though they technically are: we believe they will usually be present in the context where `eless` would be installed.

**Reply by raxod502** [reply]
That's just because they are shipped in the base macOS distribution, no?

**Comment by kaushalmodi** [comment]
@indigoviolet I've converted this to a discussion as this isn't something that can be fixed in this repo.
